modules/internal/ChapelArray.chpl

```diff
@@ -2027,7 +2030,28 @@ module ChapelArray {
     proc tail(): this._value.eltType {
       return this[this.domain.high];
     }
-
```

Comment:
I'd like to see a comment saying something like:

Return a range that's a grow or a shrink of 'r' to accommodate 'd2'.